internal/server/authn/authn_method.go

```diff
@@ -16,9 +16,14 @@ type LoginMethod interface {
 	RequiresUpdate(db *gorm.DB) (bool, error) // Temporary way to check for one time password re-use, remove with #1441
 }
 
-func Login[T LoginMethod](db *gorm.DB, loginMethod T, keyExpiresAt time.Time, keyExtension time.Duration) (*models.AccessKey, string, error) {
+func Login(db *gorm.DB, loginMethod interface{}, keyExpiresAt time.Time, keyExtension time.Duration) (*models.AccessKey, string, error) {
```

Comment:
suggest replacing `loginMethod interface{}` with `loginMethod LoginMethod`, because you can reference the exact interface you need to call, then you don't need type assertion below.

Reply:
That makes way more sense, fixed